sql/core/src/main/scala/org/apache/spark/sql/execution/command/tables.scala

```diff
@@ -1110,7 +1110,9 @@ case class ShowCreateTableCommand(table: TableIdentifier)
         }
 
         if (tableMetadata.tableType == VIEW) {
-          throw new AnalysisException("Hive view isn't supported by SHOW CREATE TABLE")
+          throw new AnalysisException("Hive view isn't supported by SHOW CREATE TABLE. " +
```

Comment:
BTW, are all views Hive view? It looks a bit weird to treat view as Hive serde table.

Reply:
We have a `DDLUtils.isDatasourceTable` check above. Will other view come to here?

Reply:
Ideally, view is not table so SHOW CREATE TABLE should not work for views. If we don't want to add a SHOW CREATE VIEW and just want to support view in SHOW CREATE TABLE, then I think it's better to not require the AS SERDE.

Reply:
What you mean "not require the AS SERDE"? You mean "SHOW CREATE TABLE" should work on Hive view?

Reply:
IMO there is no such thing called "Hive view". Spark has permanent views which can be stored in Hive metatore, which is no different from views created by Hive.

Furthermore, what's the Hive specify syntax for creating views? Maybe we should support views in both SHOW CREATE TABLE and SHOW CREATE TABLE AS SERDE

Reply:
OK. I get your point now.

This is Hive syntax for creating views:

```
CREATE VIEW [IF NOT EXISTS] [db_name.]view_name [(column_name [COMMENT column_comment], ...) ]
  [COMMENT view_comment]
  [TBLPROPERTIES (property_name = property_value, ...)]
  AS SELECT ...;
```
https://cwiki.apache.org/confluence/display/Hive/LanguageManual+DDL#LanguageManualDDL-CreateView


This is Spark syntax for creating views:
```
CREATE [OR REPLACE] [[GLOBAL] TEMPORARY] VIEW [IF NOT EXISTS [db_name.]view_name
    create_view_clauses
    AS query;
```
https://spark.apache.org/docs/3.0.0-preview/sql-ref-syntax-ddl-create-view.html

Looks like it is basically the same. I think we could support views in both commands as you said.

Reply:
If so, then I think I could close this follow-up and prepare a full PR to add the support. WDYT?

Reply:
Btw, for a view, what `DDLUtils.isDatasourceTable`  will tell about it?

Reply:
SGTM. To check a view, the provider should be empty and the table type should be VIEW.

Reply:
`DDLUtils.isDatasourceTable` and `DDLUtils.isHiveTable` both return false for views.